libraries/AP_Filesystem/AP_Filesystem_LocationDB.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  ArduPilot filesystem interface for LocationDB 
+ */
+
+#include "AP_Filesystem_config.h"
+
+#if AP_FILESYSTEM_LOCATIONDB_ENABLED
+
+#include "AP_Filesystem.h"
+#include "AP_Filesystem_LocationDB.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_AHRS/AP_AHRS.h>
+
+extern const AP_HAL::HAL& hal;
+extern int errno;
+
+#define IDLE_TIMEOUT_MS 30000
+
+int AP_Filesystem_LocationDB::open(const char *fname, int flags, bool allow_absolute_paths)
+{
+    auto *db = AP::locationdb();
+    if (!check_file_name(fname) || !db) {
+        errno = ENOENT;
+        return -1;
+    }
+
+    bool readonly = ((flags & O_ACCMODE) == O_RDONLY);
```

Comment:
I think yes, we can do this but I followed AP_Filesystem_Mission. Also, what if more bits like O_WRONLY are also set? I mean, it might be due to an error but just in case.